src/HTTP/Cookies/CookieWrapper.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace ILIAS\HTTP\Cookies;
+
+use Dflydev\FigCookies\SetCookie;
+
+/**
+ * Class CookieFacade
+ *
+ * Facade class for the FigCookies SetCookie class.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @package ILIAS\HTTP\Cookies
+ * @since   5.2
+ * @version 1.0.0
+ */
+class CookieWrapper implements Cookie {
+
```

Comment:
In my opinion this isn't an immutable implementation of Cookie. In _with_ functions the property _cookie_ is changed and then returned new object. So we have a changed source object and a new instance and both are equal.